### DISCUSSION TITLE
How to create and utilize multiple Tool Tip themes?

### DISCUSSION THREAD

**Discussion by scribblecrumb** [MyreMylar/pygame_gui#380] [Q&A] [answered]
Hello! I've run into a problem where I need to have variable rect widths depending on the tool tip being displayed.  I would assume you can create object IDs for different tool tips themes, but from there how do you connect those themes to the buttons meant to display that tool tip?  I can't see any obvious way to do so and the documention doesn't seem very clear on how tool tips are connected to UIButtons.

Better yet, a way to make the rect_width dynamic just like the height, but with a max width that it stops at.

**Comment by MyreMylar** [comment, marked as answer]
Hello!

I've just added a patch to help with this in version 0.6.7.

You should now be able to do:

```
    "#my_button_1.tool_tip":
    {
        "misc":
        {
            "rect_width": "300"
        }
    }
    "#my_button_2.tool_tip":
    {
        "misc":
        {
            "rect_width": "100"
        }
    }
```

As tool tips now have the button they are generated from set as their 'parent' element, which seems appropriate to me. You can also now specify an ObjectID (including a class ID) for the tooltip directly when creating the button, if you want to do it that way, and have a whole class of tool tips be one width.

**Reply by scribblecrumb** [reply]
Wonderful! This is perfect, thanks a ton :)
